src/reporter/index.js

```diff
@@ -100,10 +102,29 @@ export default class Reporter {
             // fixture, we can report this fixture start.
             nextReportItem = this.reportQueue[0];
 
-            await this.plugin.reportTestDone(reportItem.test.name, reportItem.testRunInfo, reportItem.test.meta);
-
-            if (nextReportItem && nextReportItem.fixture !== currentFixture)
-                await this.plugin.reportFixtureStart(nextReportItem.fixture.name, nextReportItem.fixture.path, nextReportItem.fixture.meta);
+            await this.dispatchToPlugin({
+                method: ReporterPluginMethod.reportTestDone,
+                args:   [
+                    reportItem.test.name,
+                    reportItem.testRunInfo,
+                    reportItem.test.meta
+                ]
+            });
+
+            if (!nextReportItem)
```

Comment:
Why do we need to separate these conditions?
I would still prefer to extract the body of loop into the separate method and replace `continue` with `return`

Reply:
Optional chaining can be used:
```js
 if (nextReportItem?.fixture !== currentFixture) {
```

Reply:
I could not came up with a good name for the loop body method.
```js
async _shiftReportQueue (reportItem) {
    let currentFixture = null;
    let nextReportItem = null;

    while (this.reportQueue.length && this.reportQueue[0].testRunInfo)
        this._processReportQueue(reportItem);
}
```
`_shiftReportQueue` and `_processReportQueue` most nearly the same in meaning

Reply:
> Optional chaining can be used:
> 
> ```js
>  if (nextReportItem?.fixture !== currentFixture) {
> ```

please use this approach

Reply:
I missed something about optional chaining. Syntax sugar should be applied really carefully :-)
```js
nextReportItem?.fixture !== currentFixture
```
is being compiled into 
```js
(nextReportItem === null || nextReportItem === void 0 ? void 0 : nextReportItem.fixture) !== currentFixture
```
which is not equal to the condition
```js
nextReportItem && nextReportItem.fixture !== currentFixture
```
@AlexKamaev, should I return to the initial approach afterall? I think that it's the best one. In this case, we probably don't even need to extract the body of the loop into a separate method.

Reply:
@wentwrong it's up to you. If you think your approach is the best one you can use it.